contracts/eos/BancorConverter/migrate.cpp

```diff
@@ -0,0 +1,79 @@
+[[eosio::action]]
+void BancorConverter::migrate( const set<symbol_code> converters )
+{
+    for ( const symbol_code symcode : converters ) {
+        const symbol currency = migrate_converter( symcode );
+        migrate_reserve( currency );
+    }
+}
+
+symbol BancorConverter::migrate_converter( const symbol_code symcode )
+{
+    // tables
+    converters converters_table_scope( get_self(), symcode.raw() );
+    converters converters_table_no_scope( get_self(), get_self().value );
+
+    // iterators
+    auto converter_itr_scope = converters_table_scope.find( symcode.raw() );
+    auto converter_itr_no_scope = converters_table_no_scope.find( symcode.raw() );
+
+    // scoped symbol code must exists
+    check( converter_itr_scope != converters_table_scope.end(), symcode.to_string() + " currency symbol code does not exists" );
+
+    // create
+    if ( converter_itr_no_scope == converters_table_no_scope.end() ) {
+        // add v1 converters to main scope
+        converters_table_no_scope.emplace(get_self(), [&](auto& row) {
+            row.currency = converter_itr_scope->currency;
+            row.owner = converter_itr_scope->owner;
+            row.stake_enabled = converter_itr_scope->stake_enabled;
+            row.fee = converter_itr_scope->fee;
+        });
+    // modify
+    } else {
+        converters_table_no_scope.modify(converter_itr_no_scope, get_self(), [&](auto& row) {
```

Comment:
does that mean we can migrate the same converter multiple times with no side effects?

Reply:
Yes, you'd be able to migrate the same converter over and over without any side effects.

So if you attempt to migrate BNT twice, it won't fail or have any negative side effects